src/Parsers/ParserCreateQuery.cpp

```diff
@@ -539,21 +539,40 @@ bool ParserCreateTableQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expe
         if (!s_rparen.ignore(pos, expected))
             return false;
 
-        auto storage_parse_result = storage_p.parse(pos, storage, expected);
+        storage_p.parse(pos, storage, expected);
 
-        if (storage_parse_result && s_as.ignore(pos, expected))
+        if (storage && s_as.ignore(pos, expected))
         {
             if (!select_p.parse(pos, select, expected))
                 return false;
         }
 
-        if (!storage_parse_result && !is_temporary)
+        /// Omit ENGINE:
+        /// "CREATE TABLE [table_name] (col1 Int32, col2 Int32)" is legal
+        /// "CREATE TABLE [table_name] (col1 Int32, col2 Int32) AS SELECT ..." is legal
+        if (!storage && !is_temporary)
         {
-            if (!s_as.ignore(pos, expected))
-                return false;
-            if (!table_function_p.parse(pos, as_table_function, expected))
+            bool engine_omitted = true;
+            if (s_as.ignore(pos, expected))
             {
-                return false;
+                /// If followed by a SELECT subquery, must parse it.
+                if (select_p.parse(pos, select, expected))
+                    engine_omitted = true;
+                /// If followed by table_function, ENGINE can not be specified
+                else if (table_function_p.parse(pos, as_table_function, expected) && !is_temporary)
+                    engine_omitted = false;
+            }
+
+            if (engine_omitted)
+            {
+                storage = std::make_shared<ASTStorage>();
+                auto engine = std::make_shared<ASTFunction>();
+                engine->no_empty_args = true;
+                engine->name = "MergeTree";
+                storage->set(storage->as<ASTStorage>()->engine, engine);
+                auto order_by = std::make_shared<ASTFunction>();
+                order_by->name = "tuple";
+                storage->set(storage->as<ASTStorage>()->order_by, order_by);
```

Comment:
It's questionable. I don't think that it's good idea to substitute `ORDER BY tuple()` by default, because table will not have primary key and SELECTs will always do full scan that may be extremely slow. 

Empty primary key is usually a misconfiguration and it's not possible to change `ORDER BY` after table creation, so we definitely should not create a table with empty primary key if `ORDER BY` is not specified explicitly. It's better to throw an exception that explains that `MergeTree` engine requires `ORDER BY` key and makes a user to think about what they are trying to do.

Since we cannot automatically choose `ORDER BY` key we also cannot substitute `MergeTree` as table engine by default. So I'd prefer to close this PR.